frontend/catalyst/device/qjit_device.py

```diff
@@ -35,6 +35,7 @@
     catalyst_decompose,
     measurements_from_counts,
 )
+from catalyst.jax_tracer import get_device_shots
```

Comment:
It's not clear to me immediately why `get_device_shots` is something that should be defined in the `jax_tracer` module, rather than something relating to devices 🤔

Reply:
It was mostly to avoid circular dependencies. I'll move it to `qjit_device.py` and if that works I'll keep it there.

Reply:
There was a circular dependency, so I duplicated the helper in `jax_tracer.py` as well.